lldb/source/Plugins/ObjectFile/ELF/ObjectFileELF.cpp

```diff
@@ -3805,6 +3850,28 @@ DataExtractor ObjectFileELF::GetSegmentData(const ELFProgramHeader &H) {
   return DataExtractor();
 }
 
+DataExtractor ObjectFileELF::GetSectionData(const elf::ELFSectionHeader &sh) {
+  // Try and read the section contents from our cached m_data which can come
+  // from the file on disk being mmap'ed or from the initial part of the ELF
+  // file we read from memory and cached.
+  DataExtractor data;
+  if (GetSectionContentsFromELFData(sh, m_data, data))
+    return data;
+  if (IsInMemory()) {
+    // We have a ELF file in process memory, read the program header data from
+    // the process.
+    if (ProcessSP process_sp = m_process_wp.lock()) {
+      const addr_t data_addr = m_memory_addr + sh.sh_offset;
+      if (DataBufferSP data_sp = ReadMemory(process_sp, data_addr, sh.sh_size)) {
+        data = DataExtractor(data_sp, GetByteOrder(), GetAddressByteSize());
+        if (data.GetByteSize() == sh.sh_size)
```

Comment:
Wouldn't it be better to check the content of `data_sp` before copying it to the DataExtractor and then checking if it's the right size?

So 
```
if (DataBufferSP data_sp = ReadMemory(process_sp, data_addr, sh.sh_size)) {
     if (data_sp.GetByteSize() == sh.sh_size) 
          return DataExtractor(data_sp, GetByteOrder(), GetAddressByteSize());
```